src/Common/tests/System/PlatformDetection.cs

```diff
@@ -456,5 +456,28 @@ public bool IsInRange(Version version)
                 return version >= Start && (Finish == null || version <= Finish);
             }
         }
+
+        public static bool IsEnvironmentCommandLineArgsSupported => s_IsEnvironmentCommandLineArgsSupported.Value;
+
+        private static readonly Lazy<bool> s_IsEnvironmentCommandLineArgsSupported = new Lazy<bool>(
+            delegate ()
+            {
+                if (!PlatformDetection.IsNetNative)
+                    return true;
+
+                try
+                {
+                    Environment.GetCommandLineArgs();
+                }
+                catch (PlatformNotSupportedException)
+                {
+                    return false;
+                }
+                catch
+                {
+                    return true; // Hmm - threw something but not PNSE. Let the tests loose on it and report what they find...
```

Comment:
I think the more general concept deserves an issue of its own: How do we arrange things so that test disablers that no longer fire get killed off so they never reawaken again?

Causing it to Assert() immediately seems too draconian (implement an api in one repo - boom! new test failures in the other repo as soon as it propagates.)

What we need is some sort of "Warning Assert" - i.e. something that lets us know there's a test cleanup to be done but doesn't need to be done immediately or cause a test failure...